### ISSUE TITLE
Quick Start only works when sending and receiving Events, but doesn't work when just receiving

### ISSUE THREAD

**Issue by heli567** [Azure/azure-event-hubs-go#206]
### Expected Behavior
The event hub handler should receive events with lastest offset from the EventHub.

### Actual Behavior
After adding the "eventhub.ReceiveWithLatestOffset()", the code read nothing back from Event Hub.

hub.Receive(ctx, partitionID, handler, eventhub.ReceiveWithLatestOffset(), eventhub.ReceiveWithConsumerGroup("$Default"))

(I tried to remove "eventhub.ReceiveWithLatestOffset()" and it will read back all the events from the Event Hub.)

`connStr := "Endpoint=xxxxxxxxxx"
	hub, err := eventhub.NewHubFromConnectionString(connStr)

	if err != nil {
		fmt.Println(err)
		return
	}

	ctx, cancel := context.WithTimeout(context.Background(), 2000*time.Second)
	defer cancel()

	// send a single message into a random partition
	err = hub.Send(ctx, eventhub.NewEventFromString("hello, world!"))
	if err != nil {
		fmt.Println(err)
		return
	}
	fmt.Println("send success!!!")

	handler := func(c context.Context, event *eventhub.Event) error {
		fmt.Println("start handler")
		fmt.Println(string(event.Data))
		fmt.Println("end handler")
		return nil
	}

	// listen to each partition of the Event Hub
	runtimeInfo, err := hub.GetRuntimeInformation(ctx)
	if err != nil {
		fmt.Println(err)
		return
	}
	
	for _, partitionID := range runtimeInfo.PartitionIDs {
		// Start receiving messages 
		// 
		// Receive blocks while attempting to connect to hub, then runs until listenerHandle.Close() is called 
		// <- listenerHandle.Done() signals listener has stopped
		// listenerHandle.Err() provides the last error the receiver encountered 
		listenerHandle, err := hub.Receive(ctx, partitionID, handler, eventhub.ReceiveWithLatestOffset(), eventhub.ReceiveWithConsumerGroup("$Default"))
		if err != nil {
			fmt.Println(err)
			return
		}
		fmt.Println(listenerHandle.Err())
    }

	// Wait for a signal to quit:
	signalChan := make(chan os.Signal, 1)
	signal.Notify(signalChan, os.Interrupt, os.Kill)
	<-signalChan

	err = hub.Close(context.Background())
	if err != nil {
		fmt.Println(err)
	}`

### Environment
- OS: mac os 10.15.7
- Go version: go1.15 darwin/amd64
- Version of Library: github.com/Azure/azure-event-hubs-go/v3 v3.3.7

**Comment by mayurid** [comment]
@jhendrixMSFT and @catalinaperalta : Could you please take a look

**Comment by catalinaperalta** [comment]
Thanks for reaching out! Will look into this and update this issue with any findings I have.

**Comment by catalinaperalta** [comment]
I've confirmed that there is a problem when attempting to receive with `eventhub.ReceiveWithLatestOffset()`. Looking into where the bug is.

**Comment by catalinaperalta** [comment]
I believe I have good news on this issue. It seems that the Event Hubs service has had some performance improvements which pointed out an issue with our quick start. 

When using `eventhub.ReceiveWithLatestOffset()`, this will create a new link that will only receive events after the time of the latest offset with which it is set up. So there was an unintentional dependency between the sending and receiving side in the quick start. The previous behavior probably counted on Event Hubs being a little slower in sending the first event, so that the link could be set up and read the event which was sent afterwards. However now, the link is set up with the latest offset but the event was sent before it therefore you arent seeing anything being received. To test out the functionality with `eventhub.ReceiveWithLatestOffset()`, I suggest having a separate producer which is sending events after the link is set up. You can use the producer in our additional samples [here](https://github.com/Azure/azure-event-hubs-go/tree/master/_examples/helloworld).

I am adding a PR to fix the readme to remove `eventhub.ReceiveWithLatestOffset()` from the Receive() call.

Hope that cleared up the problem!
